packages/embark-transaction-manager/src/lib/index.ts

```diff
@@ -0,0 +1,85 @@
+import { Embark, Events, Logger } /* supplied by @types/embark in packages/embark-typings */ from "embark";
+import { AccountParser, dappPath } from "embark-utils";
+import { EventEmitter } from "events";
+import Web3 from "web3";
+import EthAccounts from "./eth_accounts";
+import EthSendTransaction from "./eth_sendTransaction";
+import EthSignTypedData from "./eth_signTypedData";
+import PersonalNewAccount from "./personal_newAccount";
+import BaseTransaction from "./Transaction";
+
+export default class TransactionManager {
+
+  private modifiers: BaseTransaction[] = [];
+  private _web3: Web3 | null = null;
+  private txEvents: EventEmitter;
+  private logger: Logger;
+  private events: Events;
+  public _accounts: any[] | null = null;
+  public _nodeAccounts: any[] | null = null;
+  constructor(private readonly embark: Embark) {
+    this.events = embark.events;
+    this.logger = embark.logger;
+    this.txEvents = new EventEmitter();
+    this.init();
+  }
+
+  private async init() {
+    const web3 = await this.web3;
+    const nodeAccounts = await this.nodeAccounts;
+    const accounts = await this.accounts;
+    this.modifiers = [PersonalNewAccount, EthAccounts, EthSendTransaction, EthSignTypedData].map((rpcClass) => new rpcClass(this.embark, web3, nodeAccounts, accounts, this.txEvents));
+
+    this.txEvents.on("nodeAccounts:updated", async (updatedNodeAccounts) => {
+      this._nodeAccounts = updatedNodeAccounts;
+      this._accounts = null;
+      const accts = await this.accounts;
+      for (const modifier of this.modifiers) {
+        modifier.accounts = accts;
+      }
+    });
+    this.txEvents.on("nodeAccounts:added", async (addedNodeAccount) => {
+      if (!this._nodeAccounts) {
+        this._nodeAccounts = [addedNodeAccount];
+      } else {
+        this._nodeAccounts.push(addedNodeAccount);
+      }
+      this._accounts = null;
+      const accts = await this.accounts;
+      for (const modifier of this.modifiers) {
+        modifier.accounts = accts;
+      }
+    });
+  }
```

Comment:
The two events above almsot have the same function content. Might be nice to find a way to merge them
Something like:
```
reprocessAccounts(updatedNodeAccounts) {
  this._nodeAccounts = updatedNodeAccounts;
      this._accounts = null;
      const accts = await this.accounts;
      for (const modifier of this.modifiers) {
        modifier.accounts = accts;
      }
}

this.txEvents.on("nodeAccounts:updated", reprocessAccounts.bind(this));
this.txEvents.on("nodeAccounts:added", async (addedNodeAccount) => {
    if (!this._nodeAccounts) {
      this._nodeAccounts = [addedNodeAccount];
    } else {
      this._nodeAccounts.push(addedNodeAccount);
    }
  reprocessAccounts(this._nodeAccounts);
});
```

Reply:
Updated.